### PR TITLE
Add spidermonkey-ufi project

### PR DESCRIPTION
**Pull request by posidron**
Adds project for Spidermonkey UFI (Universal Fuzzing Interface). LibFuzzer's which run inside of the JS Shell.

### CODE REVIEW

**Comment by inferno-chromium** [comment]
Can you please check the failure

```
tatus: Downloaded newer image for gcr.io/oss-fuzz-base/base-runner:latest
INFO: performing bad build checks for /out/StructuredCloneReader.
INFO: performing bad build checks for /out/Wasm.
Broken fuzz targets (2):
StructuredCloneReader:
BAD BUILD: /out/StructuredCloneReader seems to have either startup crash or exit:
/out/StructuredCloneReader: error while loading shared libraries: libc++.so.1: cannot open shared object file: No such file or directory
Wasm:
BAD BUILD: /out/Wasm seems to have either startup crash or exit:
/out/Wasm: error while loading shared libraries: libc++.so.1: cannot open shared object file: No such file or directory
ERROR: 100% of fuzz targets seem to be broken. See the list above for a detailed information.
Check build failed.
Failed projects: spidermonkey-ufi
The command "./infra/travis/travis_build.py" exited with 1.
```

**Comment by posidron** [comment]
@inferno-chromium I was wondering why this happens with the normal spidermonkey project as well but runs and thought this will be handled internally by the base_runner container. I will post a follow up patch using the system libs.

**Comment by inferno-chromium** [comment]
@oliverchang @jonathanmetzman should know

File: projects/spidermonkey-ufi/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# limitations under the License.
+#
+################################################################################
+
+FROM gcr.io/oss-fuzz-base/base-builder
+MAINTAINER cdiehl@mozilla.com
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    autoconf2.13 \
+    python \
+    libc++1 \
```
Comment (jonathanmetzman) [inline projects/spidermonkey-ufi/Dockerfile]: I'm not going to block this if it works. But this seems like a bit of a crazy hack to me (though I don't really know much about linking).
Is it not possible to build these fuzzers with libc++ linked statically? No other project seems to have this issue.

If not, we may be able to solve this on the oss-fuzz side by building the shared versions of libc++ and libc++abi (which we explicitly opt-out of [here](https://github.com/google/oss-fuzz/blob/master/infra/base-images/base-clang/checkout_build_install_llvm.sh#L86)). So you don't need to install one from apt.

@morehouse WDYT about this?

**Comment by choller** [comment]
I don't think we support linking with libc++ statically in either Firefox or the JS shell. And the fuzzing targets are not separate from that.

**Pull request merged**